<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Serial</class>
 <widget class="QMainWindow" name="Serial">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1208</width>
    <height>756</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Serial</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_9">
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="minimumSize">
       <size>
        <width>690</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>发送区</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QTabWidget" name="tabWidget">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="tab_3">
          <attribute name="title">
           <string>文本发送区</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="QLineEdit" name="edit_send"/>
               </item>
               <item>
                <widget class="QPushButton" name="btn_send">
                 <property name="text">
                  <string>发送</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPlainTextEdit" name="edit_text_buffer">
               <property name="sizeAdjustPolicy">
                <enum>QAbstractScrollArea::AdjustIgnored</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tab_4">
          <attribute name="title">
           <string>数据展示区</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QPushButton" name="btn_draw">
               <property name="text">
                <string>开始绘图</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_stop">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>停止绘图</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="1" column="0">
            <widget class="QChartView" name="chart_view"/>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="maximumSize">
       <size>
        <width>350</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>设置区</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QComboBox" name="cmb_com">
             <item>
              <property name="text">
               <string>COM1</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>COM2</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>COM3</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>COM4</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>COM5</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>COM6</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>COM7</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>COM8</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>COM9</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="cmb_baud">
             <item>
              <property name="text">
               <string>115200</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>57600</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>38400</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>19200</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>9600</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>4800</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>2400</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>1200</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="cmb_parity">
             <item>
              <property name="text">
               <string>none</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>even</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>odd</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>mark</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>space</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="cmb_data_bits">
             <item>
              <property name="text">
               <string>8</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>7</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>6</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>5</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="cmb_stop_bits">
             <item>
              <property name="text">
               <string>1</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>1.5</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>2</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QGroupBox" name="groupBox_3">
             <property name="title">
              <string>发送进制选择</string>
             </property>
             <widget class="QRadioButton" name="rad_hex">
              <property name="geometry">
               <rect>
                <x>10</x>
                <y>30</y>
                <width>51</width>
                <height>19</height>
               </rect>
              </property>
              <property name="text">
               <string>HEX</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
             <widget class="QRadioButton" name="rad_oct">
              <property name="geometry">
               <rect>
                <x>60</x>
                <y>30</y>
                <width>51</width>
                <height>19</height>
               </rect>
              </property>
              <property name="text">
               <string>OCT</string>
              </property>
             </widget>
             <widget class="QRadioButton" name="rad_bin">
              <property name="geometry">
               <rect>
                <x>120</x>
                <y>30</y>
                <width>51</width>
                <height>19</height>
               </rect>
              </property>
              <property name="text">
               <string>BIN</string>
              </property>
             </widget>
             <widget class="QRadioButton" name="rad_dec">
              <property name="geometry">
               <rect>
                <x>170</x>
                <y>30</y>
                <width>51</width>
                <height>19</height>
               </rect>
              </property>
              <property name="text">
               <string>DEC</string>
              </property>
             </widget>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>周期：</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="edit_circle">
             <property name="placeholderText">
              <string>ms 单位，默认1000 ms</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="chk_circle">
             <property name="text">
              <string> 循环</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>数据路径：</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="edit_data_path">
             <property name="placeholderText">
              <string> 例：C:/test/</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QCheckBox" name="chk_txt">
             <property name="text">
              <string>TXT</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="chk_csv">
             <property name="text">
              <string>CSV</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="chk_jpg">
             <property name="text">
              <string>JPG</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QPushButton" name="btn_open">
             <property name="text">
              <string>连接端口</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_clear">
             <property name="text">
              <string>清空发送区</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_close">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>关闭端口</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QPushButton" name="btn_ok">
             <property name="text">
              <string>确定</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_reset">
             <property name="text">
              <string>重置</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_save">
             <property name="text">
              <string>保存</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QChartView</class>
   <extends>QGraphicsView</extends>
   <header location="global">qchartview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
